juju/model.py

```diff
@@ -1684,6 +1690,38 @@ async def _resolve_architecture(self, url):
 
         return DEFAULT_ARCHITECTURE
 
+    async def _add_charmhub_resources(self, application,
+                                      entity_url,
+                                      origin,
+                                      overrides=None):
+        charm_facade = client.CharmsFacade.from_connection(self.connection())
+        res = await charm_facade.CharmInfo(entity_url)
+
+        resources = []
+        for resource in res.meta.resources.values():
+            resources.append({
+                'description': resource.description,
+                'name': resource.name,
+                'path': resource.path,
+                'type_': resource.type_,
+                'origin': 'store',
+                'revision': -1,
```

Comment:
We don't get provided with any resource revisions in the facade call.

Reply:
The resource revision is optional and provided by the user on the cli.  Just like they can upload a resource file by specifying it, rather than use the version in the store.  Which is missing here too.

https://github.com/juju/juju/blob/7a0b2cd2ebf913530eb97d26f76628d4d8e30c33/resource/resourceadapters/deploy.go#L55

Check out the juju-qa-test charm, it has non k8s resources, you can specify by revision, or upload a file, or neither.